executor.go

```diff
@@ -82,22 +82,13 @@ func (e *Executor) Config() *Config { return e.config }
 // Subscribe returns a channel of executor-level events. Each invocation
 // of Events() returns a new channel. The done function should be called
 // to unregister this channel.
-func (e *Executor) Subscribe() (events <-chan ExecutorEvent, done func()) {
+func (e *Executor) Subscribe() (events <-chan ExecutorEvent) {
 	e.subscriptionsMu.Lock()
 	defer e.subscriptionsMu.Unlock()
 
-	once := sync.Once{}
-
 	ch := make(chan ExecutorEvent, 1024)
 	e.eventsChs[ch] = struct{}{}
-	return ch, func() {
-		once.Do(func() {
-			e.subscriptionsMu.Lock()
-			defer e.subscriptionsMu.Unlock()
-			close(ch)
-			delete(e.eventsChs, ch)
```

Comment:
do we need to clean this up still? otherwise can the executor try to write to the closed channel?

Reply:
a description for before/after on stopping the executor + reporters would be helpful

Reply:
👍 I kind of want to pull this together with the other refactors - do you mind if I merge the hackathon work into this PR?

Reply:
@stephen I've moved this into a seaparate PR